Sources/Interaction/Manipulators/MouseCameraUnicamManipulator/index.js

```diff
@@ -0,0 +1,311 @@
+      offset[index] = delta.x * rightV[index] + delta.y * upV[index];
+    }
+
+    camera.translate(...offset);
+
+    renderer.resetCameraClippingRange();
+    interactor.render();
+  };
+
+  //----------------------------------------------------------------------------
+  const dolly = (interactor, position) => {
+    const renderer = interactor.findPokedRenderer();
+    const normalizedPosition = normalize(position, interactor);
+    const normalizedPreviousPosition = normalize(
+      model.previousPosition,
+      interactor
+    );
+
+    const delta = {
+      x: normalizedPosition.x - normalizedPreviousPosition.x,
+      y: normalizedPosition.y - normalizedPreviousPosition.y,
+    };
+
+    const camera = renderer.getActiveCamera();
+    const cameraPosition = camera.getPosition();
+
+    // 1. Handle dollying
+    if (camera.getParallelProjection()) {
+      camera.zoom(1 - delta.y);
+    } else {
+      const offset1 = [];
+      vtkMath.subtract(model.downPoint, cameraPosition, offset1);
+      vtkMath.multiplyScalar(offset1, delta.y * -4);
+
+      camera.translate(...offset1);
+    }
+
+    // 2. Now handle side-to-side panning
+    const { rightV: offset2 } = getRightVAndUpV(model.downPoint, interactor);
+
+    vtkMath.multiplyScalar(offset2, delta.x);
+
+    camera.translate(...offset2);
+
+    renderer.resetCameraClippingRange();
+    interactor.render();
+  };
+
+  //----------------------------------------------------------------------------
+  // Public API methods
+  //----------------------------------------------------------------------------
+  publicAPI.onButtonDown = (interactor, renderer, position) => {
+    model.buttonPressed = true;
+    model.startPosition = position;
+    model.previousPosition = position;
+    model.time = Date.now() / 1000.0;
+    model.dist = 0;
+
+    model.rotateManipulator.onButtonDown(interactor, renderer, position);
+
+    model.downPoint = model.rotateManipulator.getDownPoint();
+  };
+
+  //----------------------------------------------------------------------------
+  publicAPI.onMouseMove = (interactor, renderer, position) => {
+    if (!model.buttonPressed) {
+      return;
+    }
+
+    if (model.rotateManipulator.getState() === States.IS_ROTATE) {
+      model.rotateManipulator.onMouseMove(interactor, renderer, position);
+    } else {
+      switch (model.state) {
+        case States.IS_NONE:
+          choose(interactor, position);
+          break;
+        case States.IS_PAN:
+          pan(interactor, position);
+          break;
+        case States.IS_DOLLY:
+          dolly(interactor, position);
+          break;
+        default:
+          break;
+      }
+    }
+
+    model.previousPosition = position;
+  };
+
+  //--------------------------------------------------------------------------
+  publicAPI.onButtonUp = (interactor) => {
+    model.buttonPressed = false;
+    if (model.state === States.IS_NONE) {
+      model.rotateManipulator.onButtonUp(interactor);
```

Comment:
it's too bad that rotateManipulator calls render() even if we were not in rotation mode